mlrun/projects/pipelines.py

```diff
@@ -793,43 +852,12 @@ def run(
         pipeline_context.clear()
         return _PipelineRunStatus(
             run_id,
-            cls,
+            inner_engine,
             project=project,
             workflow=workflow_spec,
             state=state,
-            run_object=run,
         )
 
-    @staticmethod
-    def wait_for_completion(
-        run_id, project=None, timeout=None, expected_statuses=None, run_object=None
-    ):
-        # Note: here the run parameter is a RunObject
-        state = run_object.wait_for_completion(timeout=timeout)
-        if state == mlrun.runtimes.constants.RunStates.completed:
-            return mlrun.run.RunStatuses.succeeded
-        return mlrun.run.RunStatuses.failed
-
-    @staticmethod
-    def get_run_status(
-        project,
-        run,
-        timeout=None,
-        expected_statuses=None,
-        notifiers: mlrun.utils.notifications.CustomNotificationPusher = None,
-    ):
-        if timeout is None:
-            timeout = 60 * 60
-        # Note: here the run parameter is _PipelineRunStatus
-        # Watching inner workflow:
-        inner_engine_kind = run.run_object.status.results.get("engine", None)
-        inner_engine = get_workflow_engine(inner_engine_kind)
-        run._engine = inner_engine
-        inner_engine.get_run_status(project=project, run=run, timeout=timeout)
-        run._engine = _RemoteRunner
-        # Watching load_and_run function:
-        run.wait_for_completion(timeout=timeout)
```

Comment:
I guess this is from your old implementation, why is this redundant now?

Reply:
The `_RemoteRunner` now returns in his `run` method the status with the **inner engine** which makes the code cleaner. In this way we can simply remove `wait_for_completion` and `get_run_status` from the `remote` engine. and the relevant function will be called automatically.